### PR TITLE
Added VCF-dumper readme file

### PR DESCRIPTION
**Pull request by pabarcgar**

### CODE REVIEW

File: vcf-dumper/README.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        <eva.mongo.collections.variants>variants</eva.mongo.collections.variants>
+        <eva.mongo.collections.files>files</eva.mongo.collections.files>
+    </properties>
+</profile>
+```
+
+To build the project artifacts, execute `mvn package -P *chosen-profle*` in the vcf-dumper directory. An executable jar will be generated in *vcf-dumper-cli/target* and a war file in *vcf-dumper-ws/target*. The war file has been tested successfully in Apache Tomcat 9.
+
+## Test
+In order to test the VCF dumper, a MongoDB server containing variants in [EVA format](https://github.com/EBIvariation/eva-pipeline/wiki/MongoDB-schema "EVA MongoDB schema") is needed. Some small test databases dumps are provided in the [test resouces](vcf-dumper-lib/src/test/resources/dump). Those dumps can be imported to a MongoDB server using the `mongorestore` command. **Those databases can be deleted if the tests are executed** (e.g., by `maven package`). There are several solutions for this: rename those databases, skip the tests when executing maven or execute *mavenrestore* after building the artifacts.
```
Comment (jmmut) [inline vcf-dumper/README.md]: more unused text "EVA MongoDB schema"

File: vcf-dumper/README.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+The Command Line Interface is intended to dump whole studies in VCF format. It does not allow filtering by region, so all the variants in the study will be dumped. This may be a time consuming operation, specially for big studies. For that reason, this tool is thought to be used only internally by EVA.
+
+This command line interface includes the following parameters:
+* **species**: species the data are associated to
+* **database**: name of the database to extract data from
+* **outdir**: output directory
+* **studies**: comma separated list of studies to query
+* **files**: comma separated list of files to query (each study in EVA can be composed by one or many files, as described [here](https://github.com/EBIvariation/eva-pipeline/wiki/MongoDB-schema#files)
+
+A succesfull command execution will produce a VCF file in the output directory.
```
Comment (jmmut) [inline vcf-dumper/README.md]: typo. change to "successful"

File: vcf-dumper/README.md [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+> db.variants.find({"chr":"20", "files.sid":"7"},{"start":1}).sort({"start":-1}).limit(1)
+{ "_id" : "20_71822_C_G", "start" : 71822 }
+> db.variants.find({"chr":"22", "files.sid":"7"},{"start":1}).sort({"start":1}).limit(1)
+{ "_id" : "22_16050075_A_G", "start" : 16050075 }
+> db.variants.find({"chr":"22", "files.sid":"7"},{"start":1}).sort({"start":-1}).limit(1)
+{ "_id" : "22_16110950_G_T", "start" : 16110950 }
+```
+We can see that the study *7* has variants in the chromosomes *20* (between position 60343 to 71822) and *22* (between 16060075 and 16110950). 
+
+Now we have enough information for run some queries:
```
Comment (jmmut) [inline vcf-dumper/README.md]: I would say "to run" or "for running". native help @tomdcsmith !

File: vcf-dumper/README.md [hunk 4]
```diff
@@ -0,0 +1,122 @@
+# European Variation Archive (EVA) VCF dumper
+
+The VCF dumper is a web service that supports some queries from the [EVA core REST Webservices API](https://github.com/EBIvariation/eva-ws/wiki#variants "EVA WS Variants endpoints") and writes the output in [VCF format](https://samtools.github.io/hts-specs/VCFv4.3.pdf "VCF format specification")
```
Comment (pabarcgar) [inline vcf-dumper/README.md]: This should be showed as a "popup" when posining the ctioing te cursor over the link, without clicking it.

File: vcf-dumper/README.md (hunk 1 above)
Comment (pabarcgar) [inline vcf-dumper/README.md]: The same that the previous comment

File: vcf-dumper/README.md (hunk 4 above)
Comment (cyenyxe) [inline vcf-dumper/README.md]: If there is already the paragraph text I don't see a big point in adding a popup...

File: vcf-dumper/README.md (hunk 3 above)
Comment (tomdcsmith) [inline vcf-dumper/README.md]: Either is fine, personally I slightly prefer "for running" in this instance, but it doesn't make much difference.

File: vcf-dumper/README.md [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+    </properties>
+</profile>
+```
+
+To build the project artifacts, execute `mvn package -P *chosen-profle*` in the vcf-dumper directory. An executable jar will be generated in *vcf-dumper-cli/target* and a war file in *vcf-dumper-ws/target*. The war file has been tested successfully in Apache Tomcat 9.
+
+## Test
+In order to test the VCF dumper, a MongoDB server containing variants in [EVA format](https://github.com/EBIvariation/eva-pipeline/wiki/MongoDB-schema "EVA MongoDB schema") is needed. Some small test databases dumps are provided in the [test resouces](vcf-dumper-lib/src/test/resources/dump). Those dumps can be imported to a MongoDB server using the `mongorestore` command. **Those databases can be deleted if the tests are executed** (e.g., by `maven package`). There are several solutions for this: rename those databases, skip the tests when executing maven or execute *mavenrestore* after building the artifacts.
+
+Once we got a server with data, and a jar or war artifact pointing to it, we can try some queries.
```
Comment (tomdcsmith) [inline vcf-dumper/README.md]: "Once we have"

File: vcf-dumper/README.md [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+
+### Queries
+The VCF dumper has two main interfaces that can be used to execute queries over the archive and get the result in VCF format:
+* **WebServices**: allows to dump the variants of some study(ies) that are in a given genomic region (or list of regions). Queries with no region are not allowed
+* **Command Line Interface**: for EVA internal use only. Allows to dump all the variants for a given study(ies) and file(s). It does not include region filters
+
+#### Webservices
+The Webservices API contains actually a single endpoint: `{baseURL}/{regionId}/variants`.
+{regionId} can be a single region or a comma separated list of regions. Each region is a composed by a chromosome name, and optionally a ':' character followed by natural numbers pair (start and end), separated by '-'. Some examples of valid regions are:
```
Comment (tomdcsmith) [inline vcf-dumper/README.md]: "Each region is composed of a chromosome..."

File: vcf-dumper/README.md [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+* annot-ct
+* maf
+* polyphen
+* sift
+* ref
+* alt
+* miss_alleles
+* miss_gts
+
+Any time a valid call is invoqued, the client will receive a VCF file stream, containing the variants that satisfy the query criteria.
```
Comment (tomdcsmith) [inline vcf-dumper/README.md]: "invoked"

File: vcf-dumper/README.md [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+* polyphen
+* sift
+* ref
+* alt
+* miss_alleles
+* miss_gts
+
+Any time a valid call is invoqued, the client will receive a VCF file stream, containing the variants that satisfy the query criteria. 
+#### CLI
+The Command Line Interface is intended to dump whole studies in VCF format. It does not allow filtering by region, so all the variants in the study will be dumped. This may be a time consuming operation, specially for big studies. For that reason, this tool is thought to be used only internally by EVA.
```
Comment (tomdcsmith) [inline vcf-dumper/README.md]: "especially"

File: vcf-dumper/README.md [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+
+This command line interface includes the following parameters:
+* **species**: species the data are associated to
```
Comment (tomdcsmith) [inline vcf-dumper/README.md]: "associated with"

File: vcf-dumper/README.md [hunk 10]
```diff
@@ -0,0 +1,122 @@
...
+> use eva_hsapiens_test
+switched to db eva_hsapiens_test
+> db.files.find({},{sid:1, fid:1})
+{ "_id" : ObjectId("563cd52b9f44e8e80ea35828"), "fid" : "6", "sid" : "7" }
+{ "_id" : ObjectId("563cd5439f44e8e80ea35829"), "fid" : "5", "sid" : "8" }
+>
+```
+We can see in the query result that the study with id *7* is associated with the file with id *6*, and the study *8* with the file *5*. 
+
+For the regions filter, we need to know which chromosomes have variants in the database and where (position of the variants in the chromosome). As described in the [MongoDB schema wiki](https://github.com/EBIvariation/eva-pipeline/wiki/MongoDB-schema), each variant has *chr* (chromosome) and *start* (position in the chromosome where the variant start). We can figure out executing some queries in the *variants* collection:
```
Comment (tomdcsmith) [inline vcf-dumper/README.md]: "...where the variant starts."

File: vcf-dumper/README.md [hunk 11]
```diff
@@ -0,0 +1,122 @@
...
+* `1`
+* `chr1`
+* `MT`
+* `1:1000-2000`
+* `chr1:1000-2000`
+
+And an example of a region list (whole chromosome 1, chromosome 2 positions 1000 to 2000, chromosome X positions 4000 to 5000) :
+* `1,2:1000-2000,X:4000-5000`
+
+The variants endpoint also include some URL variables. Two are mandatory:
```
Comment (cyenyxe) [inline vcf-dumper/README.md]: includes

File: vcf-dumper/README.md [hunk 12]
```diff
@@ -0,0 +1,122 @@
...
+* maf
+* polyphen
+* sift
+* ref
+* alt
+* miss_alleles
+* miss_gts
+
+Any time a valid call is invoked, the client will receive a VCF file stream, containing the variants that satisfy the query criteria. 
+#### CLI
```
Comment (cyenyxe) [inline vcf-dumper/README.md]: Add blank line before this header.

File: vcf-dumper/README.md [hunk 13]
```diff
@@ -0,0 +1,122 @@
...
+The Command Line Interface is intended to dump whole studies in VCF format. It does not allow filtering by region, so all the variants in the study will be dumped. This may be a time consuming operation, especially for big studies. For that reason, this tool is thought to be used only internally by EVA.
```
Comment (cyenyxe) [inline vcf-dumper/README.md]: "command line interface" should not be capitalized.

> this tool is thought to be used only internally by EVA.

What if somebody deploys their own instance of the EVA infrastructure? We shouldn't prevent them from using the CLI.

File: vcf-dumper/README.md [hunk 14]
```diff
@@ -0,0 +1,122 @@
...
+
+This command line interface includes the following parameters:
```
Comment (cyenyxe) [inline vcf-dumper/README.md]: includes -> accepts

File: vcf-dumper/README.md [hunk 15]
```diff
@@ -0,0 +1,122 @@
...
+* **species**: species the data are associated with
+* **database**: name of the database to extract data from
+* **outdir**: output directory
+* **studies**: comma separated list of studies to query
+* **files**: comma separated list of files to query (each study in EVA can be composed by one or many files, as described [here](https://github.com/EBIvariation/eva-pipeline/wiki/MongoDB-schema#files)
```
Comment (cyenyxe) [inline vcf-dumper/README.md]: composed of

File: vcf-dumper/README.md [hunk 16]
```diff
@@ -0,0 +1,122 @@
...
+
+A successfull command execution will produce a VCF file in the output directory.
```
Comment (cyenyxe) [inline vcf-dumper/README.md]: successful

File: vcf-dumper/README.md [hunk 17]
```diff
@@ -0,0 +1,122 @@
...
+
+#### Querying the test data
+For querying the test databases we need some values for the mandatory params:
```
Comment (cyenyxe) [inline vcf-dumper/README.md]: To query the test databases, values must be assigned to the following, mandatory parameters

File: vcf-dumper/README.md [hunk 18]
```diff
@@ -0,0 +1,122 @@
...
+* species
+* studies
+* files (for CLI queries)
+* regions
+
+There are three databases in the test data: `eva_hsapiens_test`, `eva_btaurus_umd32_test` and `eva_oaries_oarv31_test`. `hsapiens_test`,  `btaurus_umd32_test` and `oaries_oarv31_test` would be valid values for the species parameter then. 
+
+For getting the files and studies ids in one database, we can query the collection *files* using the Mongo shell:
```
Comment (cyenyxe) [inline vcf-dumper/README.md]: To get the file and study IDs from a database

File: vcf-dumper/README.md [hunk 19]
```diff
@@ -0,0 +1,122 @@
...
+This command line interface includes the following parameters:
+* **species**: species the data are associated with
+* **database**: name of the database to extract data from
+* **outdir**: output directory
+* **studies**: comma separated list of studies to query
+* **files**: comma separated list of files to query (each study in EVA can be composed by one or many files, as described [here](https://github.com/EBIvariation/eva-pipeline/wiki/MongoDB-schema#files)
+
+A successfull command execution will produce a VCF file in the output directory.
+
+#### Querying the test data
```
Comment (cyenyxe) [inline vcf-dumper/README.md]: I feel this section should be placed in the wiki as it has nothing to do with the application.

File: vcf-dumper/README.md [hunk 20]
```diff
@@ -0,0 +1,122 @@
...
+> db.variants.find({"chr":"20", "files.sid":"7"},{"start":1}).sort({"start":-1}).limit(1)
+{ "_id" : "20_71822_C_G", "start" : 71822 }
+> db.variants.find({"chr":"22", "files.sid":"7"},{"start":1}).sort({"start":1}).limit(1)
+{ "_id" : "22_16050075_A_G", "start" : 16050075 }
+> db.variants.find({"chr":"22", "files.sid":"7"},{"start":1}).sort({"start":-1}).limit(1)
+{ "_id" : "22_16110950_G_T", "start" : 16110950 }
+```
+We can see that the study *7* has variants in the chromosomes *20* (between position 60343 to 71822) and *22* (between 16060075 and 16110950). 
+
+Now we have enough information for running some queries:
```
Comment (cyenyxe) [inline vcf-dumper/README.md]: information to run some queries

File: vcf-dumper/README.md [hunk 21]
```diff
@@ -0,0 +1,122 @@
...
+
+*Using the Webservices, get all the variants in some regions for the study 7:*
```
Comment (cyenyxe) [inline vcf-dumper/README.md]: in study 7

File: vcf-dumper/README.md [hunk 22]
```diff
@@ -0,0 +1,122 @@
...
+
+GET call to `{baseURL}/v1/segments/20:65000-70000,22:16080000-16100000/variants?species=hsapiens_test&studies=7`
+
+Note: for a local Tomcat running in 8080, deploying the war file produced by `maven install`, {baseURL} will be `http://localhost:8080/vcf-dumper/`
+
+*Using the CLI, get all the variants for the study 8:*
```
Comment (cyenyxe) [inline vcf-dumper/README.md]: in study 8

**Pull request merged**